### PR TITLE
ER-291 Qualifications

### PR DESCRIPTION
**Pull request by chrisjensenuk**
- Added Qualification page
- Added progressive javascript. Using jQuery
- Added TempDataExtensions so we can store complex objects in TempData

### CODE REVIEW

File: src/Employer/Employer.Web/Orchestrators/Part2/QualificationsOrchestrator.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            var weightingComparer = new WeightingComparer();
+
+            var qualifications = qualificationsToSort.OrderBy(q => q.Weighting, weightingComparer)
+                .ThenBy(q => q.QualificationType).ThenBy(q => q.Subject).ToList();
+
+            return qualifications;
+        }
+
+        private class WeightingComparer : IComparer<QualificationWeighting?>
+        {
```
Comment (sachpatel) [inline src/Employer/Employer.Web/Orchestrators/Part2/QualificationsOrchestrator.cs]: I think we should avoid using an inner class for this. Suggest moving this to domain maybe. Inspector @mgwilliam

File: src/Employer/Employer.Web/Views/Part2/Qualifications/Qualifications.cshtml [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                    {
+                        <option value="@qualificationType">@qualificationType</option>
+                    }
+                </select>
+            </div>
+
+            <div class="form-group form-group-compound">
+                <label asp-for="Subject" class="form-label-bold">Subject</label>
+                <span asp-validation-for="Subject" class="error-message"></span>
+                <input asp-for="Subject" type="text" class="form-control" maxlength="50">
```
Comment (sachpatel) [inline src/Employer/Employer.Web/Views/Part2/Qualifications/Qualifications.cshtml]: I think we decided just remove these sorts of `maxlength` attributes.

Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Views/Part2/Qualifications/Qualifications.cshtml]: @mgwilliam @leewadhams @sachpatel I know we've removed some of the HTML input validation attributes and pushing as much validation to VacancyClient as possible, but I think its worth keeping maxlength so users can't over type.

File: src/Employer/Employer.Web/Orchestrators/Part2/QualificationsOrchestrator.cs (hunk 1 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Orchestrators/Part2/QualificationsOrchestrator.cs]: @mgwilliam Happy for you to decide this one.

File: src/Employer/Employer.Web/Extensions/QualificationsExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            {
+                return new List<QualificationEditModel>();
+            }
+
+            return qualifications.Select(q => new QualificationEditModel
+            {
+                QualificationType = q.QualificationType,
+                Subject = q.Subject,
+                Grade = q.Grade,
+                Weighting = q.Weighting.ToString()
```
[the remark this answers is not in the document]
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Extensions/QualificationsExtensions.cs]: 👍 I've got a PR to remove the [Display] attributes from core and GetDisplayName will still work from the Web project.

File: src/Employer/Employer.Web/Views/Part2/Qualifications/Qualifications.cshtml (hunk 2 above)
Comment (mgwilliam) [reply src/Employer/Employer.Web/Views/Part2/Qualifications/Qualifications.cshtml]: @chrisjensenuk understood - can we expose a constant from the validation perhaps?

Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Views/Part2/Qualifications/Qualifications.cshtml]: @mgwilliam Good idea. We can do this when we revisit to add the validations.

**Pull request merged**